File: app/src/main/java/com/example/myapp/DBService/Request.java
package com.example.myapp.DBService;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "requests", indices = {@Index(value = {"location_id", "time"})})
public class Request {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    public long id;
    @ColumnInfo(name = "location_id")
    public long locationId;
    @ColumnInfo(name = "time", index = true)
    public long time;
    @ColumnInfo(name = "temperature")
    public float temperature;

    public Request(long locationId, float temperature) {
        this.locationId = locationId;
        this.temperature = temperature;
        this.time = System.currentTimeMillis();
    }

}

File: app/src/main/java/com/chplalex/shaman/mvp/model/db/Request.kt
package com.chplalex.shaman.mvp.model.db

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "requests", indices = [Index(value = ["location_id", "time"])])
class Request(
    @field:ColumnInfo(name = "location_id")
    var locationId: Long,
    @field:ColumnInfo(name = "temperature")
    var temperature: Float
) {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Long = 0
    @ColumnInfo(name = "time", index = true)
    var time: Long = System.currentTimeMillis()
}
